src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php

```diff
@@ -110,17 +130,26 @@ public function registerContainerConfiguration(LoaderInterface $loader)
      */
     public function loadRoutes(LoaderInterface $loader)
     {
-        $routes = new RouteCollectionBuilder($loader);
-        $this->configureRoutes($routes);
-        $collection = $routes->build();
+        $file = (new \ReflectionObject($this))->getFileName();
+        $kernelLoader = $loader->getResolver()->resolve($file);
+        $kernelLoader->setCurrentDir(\dirname($file));
+        $collection = new RouteCollection();
+
+        try {
+            $this->configureRoutes(new RoutingConfigurator($collection, $kernelLoader, $file, $file));
 
-        if (0 !== \count($collection)) {
-            @trigger_error(sprintf('Adding routes via the "%s:configureRoutes()" method is deprecated since Symfony 5.1 and will have no effect in 6.0; use "configureRouting()" instead.', self::class), E_USER_DEPRECATED);
+            return $collection;
+        } catch (\TypeError $e) {
+            if (0 !== strpos($e->getMessage(), sprintf('Argument 1 passed to %s::configureRoutes() must be an instance of %s,', static::class, RouteCollectionBuilder::class))) {
+                throw $e;
+            }
         }
 
-        $file = (new \ReflectionObject($this))->getFileName();
-        $this->configureRouting(new RoutingConfigurator($collection, $loader, null, $file));
+        @trigger_error(sprintf('Using type "%s" for argument 1 of method "%s:configureRoutes()" is deprecated since Symfony 5.1, use "%s" instead.', RouteCollectionBuilder::class, self::class, RoutingConfigurator::class), E_USER_DEPRECATED);
```

Comment:
as you trigger a deprecation here, shouldn't this be reporter in the UPGRADE files?

And how about doing the same for the container method ?

Reply:
changelog/upgrade files updated

> And how about doing the same for the container method ?

see the description of the PR: "let existing code in peace", there is no reason to annoy ppl with this deprecation.

Reply:
So you're deprecating here because the RouteCollection is deprecated ? Makes sense.

Reply:
As mentioned in the description also yes ;)